pinot-core/src/main/java/org/apache/pinot/core/routing/RoutingTable.java

```diff
@@ -25,16 +25,37 @@
 
 public class RoutingTable {
   private final Map<ServerInstance, List<String>> _serverInstanceToSegmentsMap;
+  private final Map<ServerInstance, Map<Integer, List<String>>> _serverInstanceToPartitionedSegmentsMap;
   private final List<String> _unavailableSegments;
   private final int _numPrunedSegments;
+  private final boolean _isPartitionRouting;
 
-  public RoutingTable(Map<ServerInstance, List<String>> serverInstanceToSegmentsMap, List<String> unavailableSegments,
-      int numPrunedSegments) {
+  public RoutingTable(Map<ServerInstance, List<String>> serverInstanceToSegmentsMap,
```

Comment:
Should we add a `PartitionRoutingTable` which extends this? One advantage of that is we can have added type-safety to ensure the right version of the routing table is passed around.